openmdao/drivers/doe_generators.py

```diff
@@ -347,26 +349,24 @@ def __call__(self, design_vars, model=None):
         # yield values for doe generated indices
         for idxs in doe.astype('int'):
             retval = []
-            var = row = 0
-            for name, meta in iteritems(design_vars):
+            row = 0
+            for var, (name, meta) in enumerate(iteritems(design_vars)):
                 size = meta['size']
                 val = np.empty(size)
                 for k in range(size):
-                    idx = idxs[var + k]
```

Comment:
This is the actual bug. `var` was the index of the design variable, but array DVs take multiple rows in `idxs`